pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -171,12 +172,16 @@ def restore_training_state(self) -> None:
         if not self._loaded_checkpoint:
             return
 
-        # restore precision plugin (scaler etc.)
-        self.trainer.precision_plugin.on_load_checkpoint(self._loaded_checkpoint)
         # restore loops and their progress
         self.restore_loops()
 
-        self.restore_optimizers_and_schedulers()
+        # restore precision plugin (scaler etc.)
+        self.trainer.precision_plugin.on_load_checkpoint(self._loaded_checkpoint)
```

Comment:
can  we move this back before restore_loops(), just in case. We don't want to change the order of things when not necessary.